firewall/prompt.go

```diff
@@ -125,9 +128,35 @@ func createPrompt(ctx context.Context, conn *network.Connection, pkt packet.Pack
 
 	// If there already is a notification, just update the expiry.
 	if n != nil {
-		n.Update(expires)
-		log.Tracer(ctx).Debugf("filter: updated existing prompt notification")
-		return
+		// Get notification state
+		n.Lock()
+		state := n.State
+		n.Unlock()
```

Comment:
copy n.SelectionActionID here as well so we can avoid re-locking the notification again below.